src/react/WithQueries.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { declareQueries } from './declareQueries';
+import { QueryResult } from '../QueryResult';
+import {
+  EnforceNonEmptyRecord,
+  ObservableQueries,
+  ProductL,
+  ProductP,
+  ProductA
+} from '../util';
+
+type QueryOutputProps<L, P> = { queries: QueryResult<L, P> };
+
+type Params<A> = A extends void
+  ? {}
+  : {
+      params: A;
+    };
+
+type Props<R extends ObservableQueries> = {
+  queries: EnforceNonEmptyRecord<R>;
+  render: React.ComponentType<QueryOutputProps<ProductL<R>, ProductP<R>>>;
+} & Params<ProductA<R>>;
+
+export function WithQueries<P extends ObservableQueries>(props: Props<P>) {
+  const WrappedComponent = React.useMemo(
```

Comment:
do we want to support mutating queries?

Reply:
not sure. Passing queries will often be done like `queries={{ q1, q2 }}` so we should have a shallow equality on the queries object. In terms of `useMemo` deps, it could be something like `Object.values(props.queries)`? If it's this easy, probably worth adding it, but it's hard to test for at the same time...

Reply:
makes sense 👍